src/Microsoft.Identity.Web/ClaimsPrincipalExtensions.cs

```diff
@@ -6,215 +6,215 @@
+			}
+
+			return tenantId;
+		}
+
+		/// <summary>
+		/// Gets the login-hint associated with a <see cref="ClaimsPrincipal"/>.
+		/// </summary>
+		/// <param name="claimsPrincipal">Identity for which to complete the login-hint.</param>
+		/// <returns>The login hint for the identity, or <c>null</c> if it cannot be found.</returns>
+		public static string? GetLoginHint(this ClaimsPrincipal claimsPrincipal)
+		{
+			return GetDisplayName(claimsPrincipal);
+		}
+
+		/// <summary>
+		/// Gets the domain-hint associated with an identity.
+		/// </summary>
+		/// <param name="claimsPrincipal">Identity for which to compute the domain-hint.</param>
+		/// <returns> The domain hint for the identity, or <c>null</c> if it cannot be found.</returns>
+		public static string? GetDomainHint(this ClaimsPrincipal claimsPrincipal)
+		{
+			if (claimsPrincipal == null)
+			{
+				throw new ArgumentNullException(nameof(claimsPrincipal));
+			}
+
+			string? tenantId = GetTenantId(claimsPrincipal);
+			string? domainHint = string.IsNullOrWhiteSpace(tenantId)
+				? null
+				: tenantId!.Equals(Constants.MsaTenantId, StringComparison.OrdinalIgnoreCase) ? Constants.Consumers : Constants.Organizations;
+
+			return domainHint;
+		}
+
+		/// <summary>
+		/// Get the display name for the signed-in user, from the <see cref="ClaimsPrincipal"/>.
+		/// </summary>
+		/// <param name="claimsPrincipal">Claims about the user/account.</param>
+		/// <returns>A string containing the display name for the user, as determined by Azure AD (v1.0) and Microsoft identity platform (v2.0) tokens,
+		/// or <c>null</c> if the claims cannot be found.</returns>
+		/// <remarks>See https://docs.microsoft.com/azure/active-directory/develop/id-tokens#payload-claims. </remarks>
+		public static string? GetDisplayName(this ClaimsPrincipal claimsPrincipal)
+		{
+			if (claimsPrincipal == null)
+			{
+				throw new ArgumentNullException(nameof(claimsPrincipal));
+			}
+
+			// Use the claims in a Microsoft identity platform token first
+			string? displayName = claimsPrincipal.FindFirstValue(ClaimConstants.PreferredUserName);
+
+			if (!string.IsNullOrWhiteSpace(displayName))
+			{
+				return displayName;
+			}
+
+			// Otherwise fall back to the claims in an Azure AD v1.0 token
+			displayName = claimsPrincipal.FindFirstValue(ClaimsIdentity.DefaultNameClaimType);
+
+			if (!string.IsNullOrWhiteSpace(displayName))
+			{
+				return displayName;
+			}
+
+			// Finally falling back to name
+			return claimsPrincipal.FindFirstValue(ClaimConstants.Name);
+		}
+
+		/// <summary>
+		/// Gets the user flow ID associated with the <see cref="ClaimsPrincipal"/>.
+		/// </summary>
+		/// <param name="claimsPrincipal">The <see cref="ClaimsPrincipal"/> from which to retrieve the user flow ID.</param>
+		/// <returns>User flow ID of the identity, or <c>null</c> if it cannot be found.</returns>
+		public static string? GetUserFlowId(this ClaimsPrincipal claimsPrincipal)
+		{
+			if (claimsPrincipal == null)
```

Comment:
All of these extension methods could be deduplicated into calling a single private shared method:

```cs
private static string? GetClaimValue(ClaimsPrincipal? claimsPrincipal, params string[] claimNames)
{
    if (claimsPrincipal == null)
    {
        throw new ArgumentNullException(nameof(claimsPrincipal));
    }

    for (var i = 0; i < claimNames.Length; i++)
    {
        var currentValue = claimsPrincipal.FindFirstValue(claimNames[i]);
        if (!string.IsNullOrEmpty(currentValue))
        {
            return currentValue;
        }
    }

    return null;
}
```